gips/data/sentinel2/sentinel2.py

```diff
@@ -910,6 +917,88 @@ def ref_geoimage(self, asset_type, sensor):
             sr_image[c] = (rad_rev_img[c] - Lu) / TLdS
         self._product_images['ref'] = sr_image
 
+
+    def _band_filename(self, band, zipfile, asset_style):
+        for name in zipfile.namelist():
+            match = re.match(self.Asset._asset_styles[asset_style]['raster-re'], name)
```

Comment:
This only works for new-style assets I think.  The original-style assets have a `{tileid}` substring so you have to `re_str.format(tileid=tile_id)`, or else you'll get the IOError every time.

The good news is that you can trash this method, I think, then use `self.metadata` instead; it's populated by the time you need it in`fmask_geoimage`; if you look in `ref_toa_geoimage` near the top you can see how filenames for particular bands are gotten, by nasty slicing into strings.

Optional:  There are only five references to `self.metadata`, so consider changing it to a dict of dicts instead of a dict of lists then index the files by band, which is what I should've done in the first place.  : P  You can even use your regex improvement in this PR to extract the band in `self.load_metadata()`, and do away with nasty slicing entirely.